src/OpenLoco/src/World/StationManager.cpp

```diff
@@ -602,9 +602,18 @@ namespace OpenLoco::StationManager
     // 0x0048F850
     static void removeStationFromCargoStats(const StationId stationId)
     {
-        registers regs;
-        regs.ebx = enumValue(stationId);
-        call(0x0048F850, regs);
+        for (auto& station : stations())
+        {
+            for (auto& stats : station.cargoStats)
+            {
+                if (stats.origin == stationId)
+                {
+                    stats.origin = StationId::null;
+                    stats.quantity = 0;
+                    Ui::WindowManager::invalidate(Ui::WindowType::station, enumValue(station.id()));
```

Comment:
Could we re-use the parameter instead of apparently recalculating it?
```suggestion
                    Ui::WindowManager::invalidate(Ui::WindowType::station, enumValue(stationId));
```

Reply:
Its a different id. `stationId` is the one we are removing from stats of `station.id()`.

Reply:
Ah, makes sense! Could we clarify the parameter name, then? e.g. `stationIdToRemove` or something.